girder/utility/s3_assetstore_adapter.py

```diff
@@ -191,6 +191,12 @@ def initUpload(self, upload):
         if upload['size'] <= 0:
             return upload
 
+        # collapse consecutive spaces in the filename into a single space
+        # this is due to a bug in S3 that does not properly handle filenames
+        # with multiple spaces in a row, resulting in a SignatureDoesNotMatch
+        # error
+        upload['name'] = ' '.join(filter(None, upload['name'].split(' ')))
```

Comment:
If they manage to get tabs or newlines in the filename, I definitely think we should replace those with spaces also, so I'm fine with \s.